packages/teuchos/core/src/Teuchos_any.hpp

```diff
@@ -198,11 +203,18 @@ class TEUCHOSCORE_LIB_DLL_EXPORT any
       return *this;
     }
 
-  //! Return true if nothing is being stored
-  bool empty() const
-    {
-      return !content;
-    }
```

Comment:
This is a break in backward compatibility and unless there is a very good reason to remove `empty()`, it should be kept.  Even if Trilinos builds does not mean that customer code will build.

Reply:
Ok, let's keep `empty` then. I've added `TEUCHOS_DEPRECATED` as you suggested.